pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -548,10 +581,14 @@ func initializeNodeAgentHandles(ctx *baseOsMgrContext) {
 	subZbootConfig.Activate()
 }
 
-func initializeZedagentHandles(ctx *baseOsMgrContext) {
+func initializeZedagentHandles(ps *pubsub.PubSub, ctx *baseOsMgrContext) {
 	// Look for BaseOsConfig , from zedagent
-	subBaseOsConfig, err := pubsublegacy.Subscribe("zedagent",
-		types.BaseOsConfig{}, false, ctx, &pubsub.SubscriptionOptions{
+	subBaseOsConfig, err := ps.NewSubscription(
+		pubsub.SubscriptionOptions{
+			AgentName:     "nodeagent",
```

Comment:
@archishou Can you redo this in a way where the agentname is correct for all the changes?
Shouldn't depend on a code reviewer and/or testing to get the basic transformation correct.

FWIW I tend to do emacs macros to automate transformations as a way to ensure it is correct. You might prefer other tools. But manual edits is certainly the wrong approach to apply code transformations like this.